event_handling/include/supplementary/NotifyTimer.h

```diff
@@ -1,13 +1,16 @@
 #pragma once
 
 #include "ITrigger.h"
+#include "engine/AlicaClock.h"
```

Comment:
This is not so cool. While the change looks nice you have a circular dependency between engine and event handling now. This class should not depend on alica-engine.